test/v1/connectors/ListBenefitsConnectorSpec.scala

```diff
@@ -61,57 +61,62 @@ class ListBenefitsConnectorSpec extends ConnectorSpec {
-    )
+          willGet(s"$baseUrl/income-tax/income/state-benefits/$nino/$taxYear", parameters = Seq("benefitId" -> benefitId)) returns Future
+            .successful(outcome)
 
-    val ifsRequestHeaders: Seq[(String, String)] = Seq(
-      "Environment"   -> "release6-environment",
-      "Authorization" -> s"Bearer release6-token"
-    )
-
-    def stubHttp(response: DownstreamOutcome[ListBenefitsResponse[HMRCStateBenefit, CustomerStateBenefit]],
-                 queryParams: Seq[(String, String)]): Unit = {
-      MockedHttpClient
-        .parameterGet(
-          url = s"$baseUrl/income-tax/income/state-benefits/$nino/$taxYear",
-          queryParams,
-          config = dummyIfsHeaderCarrierConfig,
-          requiredHeaders = ifsRequestHeaders,
-          excludedHeaders = Seq("AnotherHeader" -> "HeaderValue")
-        )
-        .returns(Future.successful(response))
-    }
+          val result = await(connector.listBenefits(request))
 
-    MockedAppConfig.ifsBaseUrl returns baseUrl
-    MockedAppConfig.ifsToken returns "release6-token"
-    MockedAppConfig.ifsEnvironment returns "release6-environment"
-    MockedAppConfig.ifsEnvironmentHeaders returns Some(allowedIfsHeaders)
-  }
+          result shouldBe outcome
+        }
+      }
 
-  "ListBenefitsConnector" when {
-    "listBenefits" when {
       "no benefitId query param is provided" must {
-        "return a 200 status for a success scenario" in new Test {
+        "return a 200 status for a success scenario" in new IfsTest with Test {
+          val request = ListBenefitsRequest(Nino(nino), TaxYear.fromMtd("2019-20"), None)
           val outcome = Right(ResponseWrapper(correlationId, validResponse))
-          stubHttp(outcome, Seq("benefitId" -> benefitId))
 
-          await(connector.listBenefits(ListBenefitsRequest(Nino(nino), taxYear, Some(benefitId)))) shouldBe outcome
+          willGet(s"$baseUrl/income-tax/income/state-benefits/$nino/$taxYear") returns Future.successful(outcome)
+
+          val result = await(connector.listBenefits(request))
+
+          result shouldBe outcome
         }
       }
+      "a benefitId is provided for a TYS tax year" must {
+        "return a 200 status for a success scenario" in new TysIfsTest with Test {
 
-      "a benefitId query param is provided" must {
-        "return a 200 status for a success scenario" in new Test {
+          val request = ListBenefitsRequest(Nino(nino), TaxYear.fromMtd("2023-24"), Some(benefitId))
           val outcome = Right(ResponseWrapper(correlationId, validResponse))
-          stubHttp(outcome, queryParams = Nil)
 
-          await(connector.listBenefits(ListBenefitsRequest(Nino(nino), taxYear, None))) shouldBe outcome
+          willGet(s"$baseUrl/income-tax/income/state-benefits/23-24/$nino", parameters = Seq("benefitId" -> benefitId)) returns Future.successful(outcome)
+
+          val result = await(connector.listBenefits(request))
+
+          result shouldBe outcome
```

Comment:
I think this spec is missing tests for "downstream returns a single error" and "downstream returns multiple errors"

Reply:
Are we sure that the error scenarios are not covered by the service tests? It seems all of the connectorSpecs in this API (bar one) ignore error scenarios and only follow happy paths. For cross-reference I took a look at property-business-api and in that one we also inconsistently sometimes test for errors and sometimes not in the connector specs. Is there a team decision on whether we should test for errors in connectorSpecs?

Reply:
yep, you're right, no need!